From 3d5277346469a543a2c93ebb9d5ce2517e46406b Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 28 Apr 2019 22:27:58 +0300
Subject: [PATCH] [SW-0] pll: use API for CPU PLL base offset

Instead of playing the include game in Makefile, use API function for
receiving the device specific CPU PLL base offset.
This is needed to bypass the lack of strcmp() function.

Note: we still play the include game.

Change-Id: I67881dc910bca028d66ed99f5fe8e2fdd88fb657
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/pll.c b/common/pll.c
index d415c53..6e26e4c 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -180,9 +180,9 @@
 	for (i = 0; i < pll_num; i++) {
 		/* DIV0 is the undivided PLL output */
 		bval = hl_pll_bypass_get(hl_pll_cfg + i, HL_PLL_DIV0);
-		if (hl_pll_cfg[i].base == CPU_PLL_BASE &&
-				bval != HL_PLL_REF_CLK &&
-				bval != HL_PLL_DIV_REF)
+		if (hl_pll_cfg[i].base == hl_pll_get_cpu_pll_base_off() &&
+		    bval != HL_PLL_REF_CLK &&
+		    bval != HL_PLL_DIV_REF)
 			continue;
 
 		if (*pll_partial_init)
diff --git a/common/pll.h b/common/pll.h
index ccfa22b..56c6641 100644
--- a/common/pll.h
+++ b/common/pll.h
@@ -60,6 +60,7 @@
 int hl_pll_init(unsigned int *pll_partial_init);
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg);
 u64 hl_pll_get_cpu_cycles_per_usec(void);
+u64 hl_pll_get_cpu_pll_base_off(void);
 int hl_pll_partial_init_get(void);
 void hl_pll_partial_init_set(void);
 
diff --git a/gaudi/gaudi.c b/gaudi/gaudi.c
index 5a25a24..cb71a2a 100644
--- a/gaudi/gaudi.c
+++ b/gaudi/gaudi.c
@@ -37,6 +37,11 @@
 	return mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS;
 }
 
+u64 hl_pll_get_cpu_pll_base_off(void)
+{
+	return mmPSOC_CPU_PLL_BASE;
+}
+
 u64 hl_pll_get_cpu_cycles_per_usec(void)
 {
 	return 120000000ULL / USEC_PER_SEC;
diff --git a/gaudi/gaudi.h b/gaudi/gaudi.h
index 8522606..44fc1c5 100644
--- a/gaudi/gaudi.h
+++ b/gaudi/gaudi.h
@@ -28,7 +28,6 @@
 #define mmNIC2_HBM_PLL_BASE		0xD78000
 #define mmNIC2_MME_PLL_BASE		0xD79000
 #define mmNIC2_TPC_PLL_BASE		0xD7A000
-#define CPU_PLL_BASE			mmPSOC_CPU_PLL_BASE
 
 #define PCI_DIV2_CLOCK			(830000000 / 5)
 #define UART_BAUDDIV			(PCI_DIV2_CLOCK / 115200)
diff --git a/goya/goya.c b/goya/goya.c
index 9e9c5a9..293b9f9 100644
--- a/goya/goya.c
+++ b/goya/goya.c
@@ -163,6 +163,11 @@
 	return mmPSOC_GLOBAL_CONF_WARM_REBOOT;
 }
 
+u64 hl_pll_get_cpu_pll_base_off(void)
+{
+	return mmCPU_PLL_BASE;
+}
+
 u64 hl_pll_get_cpu_cycles_per_usec(void)
 {
 	return 133000000ULL / USEC_PER_SEC;
diff --git a/goya/goya.h b/goya/goya.h
index 06d36f0..838aaa4 100644
--- a/goya/goya.h
+++ b/goya/goya.h
@@ -10,6 +10,5 @@
 
 #define mmCPU_PLL_BASE		0x4A2000
 #define mmIC_PLL_BASE		0x4A3000
-#define CPU_PLL_BASE		mmCPU_PLL_BASE
 
 #define NON_RST_FLOPS_3		mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3
